src/main.test.js

```diff
@@ -0,0 +1,99 @@
+const { startCreating } = require('./main');
+const { isDnaUnique, createDna  } = require('./dna');
+const { needsExclusion } = require('./exclusions');
+
+jest.mock('fs', () => {
+  const originalModule = jest.requireActual('fs');
+
+  return {
+    ...originalModule,
+    mkdirSync: () => {},
+    writeFileSync: () => {},
+    readdirSync: () => (['a.png', 'b.png']),
+  };
+});
+
+jest.mock('canvas', () => {
+  const originalModule = jest.requireActual('canvas');
+
+  return {
+    ...originalModule,
+    createCanvas: jest.fn(() => ({
+      getContext: jest.fn(() => ({
+        clearRect: jest.fn(),
+        fillRect: jest.fn(),
+        drawImage: jest.fn(),
+      })),
+      toBuffer: jest.fn(),
+    })),
+    loadImage: jest.fn(() => ({})),
+  };
+});
+
+jest.mock('./config', () => {
+  const originalConfig = jest.requireActual('./config');
+  const configOverrides = require('./__fixtures__/config_overrides');
+
+  return {
+    ...originalConfig,
+    ...configOverrides,
+  };
+});
+
+jest.mock('./dna', () => {
+  const originalModule = jest.requireActual('./dna');
+
+  return {
+    ...originalModule,
+    createDna: jest.fn(originalModule.createDna),
+    isDnaUnique: jest.fn(() => true),
+  };
+});
+
+jest.mock('./exclusions', () => ({
+  needsExclusion: jest.fn(() => false),
+}));
+
+describe('main', () => {
```

Comment:
this test file checks some generic functionality to make sure the creation process still works correctly after introducing the exclusions mechanism